pyproject.toml

```diff
@@ -1,3 +1,16 @@
 [build-system]
 requires = ["setuptools", "wheel", "oldest-supported-numpy", "packaging"]
 build-backend = "setuptools.build_meta"
+
+[tool.ruff]
+# Aim to get all pyflakes logical errors ("F") passing
+select = ["F"]
+# For now, deselect rules that do not pass
+ignore = [
+  "F401", # unused imports
+  "F541", # f-string without placeholders
+  "F632", # Comparison of literals
+  "F811", # Redefinition of unused variable
+  "F821", # Undefined name
```

Comment:
hmm this seems like quite serious error, shall we just fix it?

Reply:
Yep, I think the individual fixes should be their own PRs, otherwise this will get rather bloated!